x/compliance/keeper/msg_server_revoke_model.go

```diff
@@ -28,6 +29,17 @@ func (k msgServer) RevokeModel(goCtx context.Context, msg *types.MsgRevokeModel)
 		)
 	}
 
+	// The corresponding Model Version must be present on ledger.
+	modelVersion, found := k.modelKeeper.GetModelVersion(ctx, msg.Vid, msg.Pid, msg.SoftwareVersion)
+	if !found {
+		return nil, modeltypes.NewErrModelVersionDoesNotExist(msg.Vid, msg.Pid, msg.SoftwareVersion)
+	}
+
+	// check if softwareVersionString matches with what is stored for the given version
+	if modelVersion.SoftwareVersionString != msg.SoftwareVersionString {
+		return nil, types.NewErrModelVersionStringDoesNotMatch(msg.Vid, msg.Pid, msg.SoftwareVersion, msg.SoftwareVersionString)
+	}
+
```

Comment:
I believe we need to fix the comments on lines 46 and 92 to not mention that `Model Info is not required to be on the ledger`